lasagne/layers/conv.py

```diff
@@ -794,8 +793,9 @@ def convolve(self, input, **kwargs):
             subsample=self.stride, border_mode=border_mode,
             filter_flip=not self.flip_filters)
         output_size = self.output_shape[2:]
-        if any(s is None for s in output_size):
-            output_size = self.get_output_shape_for(input.shape, True)[2:]
+        if (not isinstance(self.output_size, T.TensorVariable) and
+                any(s is None for s in output_size)):
+            output_size = self.get_output_shape_for(input.shape)[2:]
```

Comment:
Ok. I modified your suggestion in:
```python
if isinstance(self.output_size, T.TensorVariable):
    output_size = self.output_size
elif self.output_size is None or any(s is None for s in self.output_size):
    output_size = self.get_output_shape_for(input.shape)[2:]
```
to take into account the case when `self.output_size` is `None`